internal/codeintel/codenav/syntactic.go

```diff
@@ -0,0 +1,100 @@
+package codenav
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sourcegraph/scip/bindings/go/scip"
+
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/search"
+	searchclient "github.com/sourcegraph/sourcegraph/internal/search/client"
+	"github.com/sourcegraph/sourcegraph/internal/search/result"
+	"github.com/sourcegraph/sourcegraph/internal/search/streaming"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+type candidateFile struct {
+	matches  []scip.Range // Guaranteed to be sorted
+	complete bool         // Was this file searched in its entirety, or did we hit the search count limit?
+}
+
+// findCandidateOccurrencesViaSearch calls out to Searcher/Zoekt to find candidate occurrences of the given symbol.
+// It returns a map of file paths to candidate ranges.
+func findCandidateOccurrencesViaSearch(
+	ctx context.Context,
+	client searchclient.SearchClient,
+	repo types.Repo,
+	symbol *scip.Symbol,
+	language string,
+	commit api.CommitID,
+) (map[string]candidateFile, error) {
+	var contextLines int32 = 0
+	patternType := "standard"
+	repoName := fmt.Sprintf("^%s$", repo.Name)
+	var identifier string
+	if name, ok := nameFromSymbol(symbol); ok {
+		identifier = name
+	} else {
+		return nil, errors.Errorf("can't find occurrences for locals via search")
+	}
+	// TODO: This should be dependent on the number of requested usages, with a configured global limit
+	countLimit := 500
+	searchQuery := fmt.Sprintf("type:file repo:%s rev:%s language:%s count:%d %s", repoName, string(commit), language, countLimit, identifier)
+
+	plan, err := client.Plan(ctx, "V3", &patternType, searchQuery, search.Precise, 0, &contextLines)
+	if err != nil {
+		return nil, err
+	}
+	stream := streaming.NewAggregatingStream()
+	_, err = client.Execute(ctx, stream, plan)
+	if err != nil {
+		return nil, err
+	}
+
+	results := make(map[string]candidateFile)
+	for _, match := range stream.Results {
+		fileMatch, ok := match.(*result.FileMatch)
```

Comment:
```suggestion
	for _, streamResult := range stream.Results {
		fileMatch, ok := streamResult.(*result.FileMatch)
```

The word `match` is used multiple times later, this is a bit confusing to read, because it's a different type from the `matches` slice